src/types/redis_zset.cc

```diff
@@ -930,4 +930,46 @@ rocksdb::Status ZSet::RandMember(const Slice &user_key, int64_t command_count,
   return rocksdb::Status::OK();
 }
 
+rocksdb::Status ZSet::Diff(const std::vector<Slice> &keys, MemberScores *members) {
+  std::vector<std::string> lock_keys;
```

Comment:
It should be fine to read the snapshot instead of locking keys.

Reply:
Seem other places also locked keys, so let leave as it is and fix them together.

Reply:
Hi @git-hulk, I really appreciate your feedback. I'd be happy to address this in the current PR if that works for you. Additionally, I was wondering if you could provide some more insights into the snapshotting process in kvrocks, specifically whether reading from snapshots is mutually exclusive? Your clarification would be incredibly helpful. Thanks a lot!

Reply:
Thank you. Got it.